packages/portal/src/pages/themes/_.vue

```diff
@@ -0,0 +1,175 @@
+<template>
+  <div
+    data-qa="theme page"
+    class="page white-page gridless-container responsive-font"
+  >
+    <ContentHeader
+      :title="name"
+      :description="description"
+      :media-url="shareMediaUrl"
+      button-variant="secondary"
+    />
+    <div class="divider" />
+    <EntityBadges
+      v-if="relatedTopics"
+      :title="relatedTopics.headline"
+      :entity-uris="relatedTopics.hasPart"
+      class="ml-4 mb-5"
+    />
+    <EntityCardGroup
+      v-if="relatedPersons"
+      :title="relatedPersons.headline"
+      :entity-uris="relatedPersons.hasPart"
+      card-variant="mini"
+      class="mb-4"
+      card-group-class="gridless-browse-cards"
+    />
+    <SetCardGroup
+      v-if="relatedGalleries"
+      :title="relatedGalleries.headline"
+      :set-uris="relatedGalleries.hasPart"
+      class="mb-4"
+      card-group-class="gridless-browse-cards"
+    />
+    <CallToActionBanner
+      v-if="callToAction"
+      :name="callToAction.name"
+      :text="callToAction.text"
+      :link="callToAction.relatedLink"
+      :illustration="callToAction.image"
+    />
+    <RelatedEditorial
+      v-if="entityUri"
+      :entity-uri="entityUri"
+      :card-wrapper="false"
+      :limit="6"
+      class="mb-4"
+    />
+    <section
+      v-if="curatedItems"
+      class="mb-5"
+    >
+      <h2>{{ curatedItems.headline }}</h2>
+      <ItemPreviewCardGroup
+        :items="dailySetOfCuratedItems"
+        view="grid"
+      />
+      <SmartLink
+        v-if="curatedItems.moreButton"
+        :destination="curatedItems.moreButton.url"
+        class="btn btn-outline-secondary"
+      >
+        {{ curatedItems.moreButton.text }}
+      </SmartLink>
+    </section>
+  </div>
+</template>
+
+<script>
+  import ContentHeader from '@/components/generic/ContentHeader';
+  import pageMetaMixin from '@/mixins/pageMeta';
+  import { daily } from '@/plugins/europeana/utils.js';
+
+  export default {
+    name: 'ThemePage',
+
+    components: {
+      ContentHeader,
+      CallToActionBanner: () => import('@/components/generic/CallToActionBanner'),
+      EntityBadges: () => import('@/components/entity/EntityBadges'),
+      EntityCardGroup: () => import('@/components/entity/EntityCardGroup'),
+      ItemPreviewCardGroup: () => import('@/components/item/ItemPreviewCardGroup'),
+      RelatedEditorial: () => import('@/components/related/RelatedEditorial'),
+      SetCardGroup: () => import('@/components/set/SetCardGroup'),
+      SmartLink: () => import('@/components/generic/SmartLink')
+    },
+
+    mixins: [pageMetaMixin],
+
+    async asyncData({ params, query, error, app }) {
+      const variables = {
+        locale: app.i18n.isoLocale(),
+        identifier: params.pathMatch,
+        preview: query.mode === 'preview'
+      };
+
+      try {
+        const response = await app.$contentful.query('themePage', variables);
+        const theme = response.data.data.themePage.items[0];
```

Comment:
check that an actual theme was returned, as in `theme` isn't null and/or has an `identifier`, otherwise we should show a 404 page.

Reply:
Added a check.